carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -308,41 +409,55 @@ void WMBroadcaster::addSpeedLimit(std::shared_ptr<Geofence> gf_ptr)
   for (auto el: gf_ptr->affected_parts_)
   {
     // update it with new regem
-    if (gf_ptr->max_speed_limit_->id() != lanelet::InvalId)
-    {
-      current_map_->update(current_map_->laneletLayer.get(el.id()), gf_ptr->max_speed_limit_); 
-      gf_ptr->update_list_.push_back(std::pair<lanelet::Id, lanelet::RegulatoryElementPtr>(el.id(), gf_ptr->max_speed_limit_));
-    }
-    if (gf_ptr->min_speed_limit_->id() != lanelet::InvalId)
+    if (gf_ptr->regulatory_element_->id() != lanelet::InvalId)
     {
-      current_map_->update(current_map_->laneletLayer.get(el.id()), gf_ptr->min_speed_limit_);
-      gf_ptr->update_list_.push_back(std::pair<lanelet::Id, lanelet::RegulatoryElementPtr>(el.id(), gf_ptr->min_speed_limit_));
+      current_map_->update(current_map_->laneletLayer.get(el.id()), gf_ptr->regulatory_element_);
+      gf_ptr->update_list_.push_back(std::pair<lanelet::Id, lanelet::RegulatoryElementPtr>(el.id(), gf_ptr->regulatory_element_));
     }
   }
   
 }
 
-void WMBroadcaster::addBackSpeedLimit(std::shared_ptr<Geofence> gf_ptr)
+void WMBroadcaster::addBackRegulatoryComponent(std::shared_ptr<Geofence> gf_ptr)
 {
   // First loop is to remove the relation between element and regulatory element that this geofence added initially
+  
   for (auto el: gf_ptr->affected_parts_)
   {
     for (auto regem : el.regulatoryElements())
     {
-      if (regem->attribute(lanelet::AttributeName::Subtype).value() == lanelet::DigitalSpeedLimit::RuleName)
+      // extra condition for PassingControlLine
+      if (regem->attribute(lanelet::AttributeName::Subtype).value() == lanelet::PassingControlLine::RuleName && el.isLanelet())
+      {
+        lanelet::PassingControlLinePtr pcl =  std::dynamic_pointer_cast<lanelet::PassingControlLine>(current_map_->regulatoryElementLayer.get(regem->id()));
+        // if this geofence's pcl doesn't match with the lanelets current bound side
+        bool should_process_this_regem = false;
+        for (auto control_line : pcl->controlLine())
```

Comment:
Looks like this loop is identical to the one above. Can we refactor it into a helper function?